core/src/main/java/com/datastrato/gravitino/storage/relational/JDBCBackend.java

```diff
@@ -0,0 +1,180 @@
+import com.datastrato.gravitino.storage.relational.utils.SessionUtils;
+import com.google.common.base.Preconditions;
+import java.io.IOException;
+import java.util.List;
+import java.util.Objects;
+import java.util.function.Function;
+
+/**
+ * {@link JDBCBackend} is a jdbc implementation of {@link RelationalBackend} interface. You can use
+ * a database that supports the JDBC protocol as storage. If the specified database has special SQL
+ * syntax, please implement the SQL statements and methods in MyBatis Mapper separately and switch
+ * according to the {@link Configs#ENTITY_RELATIONAL_JDBC_BACKEND_DB_TYPE_KEY} parameter.
+ */
+public class JDBCBackend implements RelationalBackend {
+
+  /** Initialize the jdbc backend instance. */
+  @Override
+  public void initialize(Config config) {
+    SqlSessionFactoryHelper.getInstance().init(config);
+  }
+
+  @Override
+  public <E extends Entity & HasIdentifier> List<E> list(
+      Namespace namespace, Entity.EntityType entityType) {
+    if (entityType == Entity.EntityType.METALAKE) {
+      List<MetalakePO> metalakePOS =
+          SessionUtils.getWithoutCommit(
+              MetalakeMetaMapper.class, MetalakeMetaMapper::listMetalakePOs);
+      return (List<E>) POConverters.fromMetalakePOs(metalakePOS);
+    } else {
+      throw new IllegalArgumentException(
+          String.format("Unsupported entity type: %s for list operation", entityType));
+    }
+  }
+
+  @Override
+  public boolean exists(NameIdentifier ident, Entity.EntityType entityType) {
+    if (entityType == Entity.EntityType.METALAKE) {
+      MetalakePO metalakePO =
+          SessionUtils.getWithoutCommit(
+              MetalakeMetaMapper.class, mapper -> mapper.selectMetalakeMetaByName(ident.name()));
+      return metalakePO != null;
+    } else {
+      throw new IllegalArgumentException(
+          String.format("Unsupported entity type: %s for exists operation", entityType));
+    }
+  }
+
+  @Override
+  public <E extends Entity & HasIdentifier> void insert(E e, boolean overwritten)
+      throws EntityAlreadyExistsException {
+    if (e instanceof BaseMetalake) {
+      MetalakePO metalakePO =
+          SessionUtils.getWithoutCommit(
+              MetalakeMetaMapper.class,
+              mapper -> mapper.selectMetalakeMetaByName(e.nameIdentifier().name()));
+      if (!overwritten && metalakePO != null) {
+        throw new EntityAlreadyExistsException(
+            String.format("Metalake entity: %s already exists", e.nameIdentifier().name()));
+      }
+      SessionUtils.doWithCommit(
+          MetalakeMetaMapper.class,
+          mapper -> {
+            if (overwritten) {
+              mapper.insertMetalakeMetaWithUpdate(POConverters.toMetalakePO((BaseMetalake) e));
+            } else {
+              mapper.insertMetalakeMeta(POConverters.toMetalakePO((BaseMetalake) e));
+            }
+          });
+    } else {
+      throw new IllegalArgumentException(
```

Comment:
We can introduce a new Exception such as `UnsupportedEntityTypeException` for a better formatter.

Reply:
Done